src/marqo/_httprequests.py

```diff
@@ -169,9 +167,20 @@ def convert_to_marqo_web_error_and_raise(response: requests.Response, err: reque
             if "limit of total fields" in reason and "exceeded" in reason:
                 raise IndexMaxFieldsError(message="Exceeded maximum number of "
                                                   "allowed fields for this index.")
+        elif open_search_error_type == "cluster_block_exception":
+            raise DiskWatermarkBreachError(message="OpenSearch cluster indexing has been blocked, most likely due to breach "
+                                           "of the `flood_stage` disk watermark. Try freeing up disk space on your OpenSearch instances. "
+                                           "For more information, please refer to cluster settings: `cluster.routing.allocation.disk.watermark` at "
+                                           "https://opensearch.org/docs/2.7/api-reference/cluster-api/cluster-settings/")
     except KeyError:
         pass
 
+    # for throttling
+    if response.status_code == 429:
+        raise TooManyRequestsError(
+            message="Marqo-OS received too many requests! "
```

Comment:
We refer to both OpenSearch as both OpenSearch and also Marqo-OS. Perhaps beyond the scope of this PR, but it's confusing

Reply:
Will make a ticket for this!